File: csharp/gol.tests/TestTheTester.cs
using System;
using Xunit;
using DeepEqual.Syntax;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace gol.tests
{
    public class TestTheTester
    {
        [Fact]
        public void ClassicWorks()
        {
            Assert.Equal(1, 1);
        }

        [Fact]
        public void DeepWorks()
        {
            var foo = new Foo { Bar = "test" };
            foo.ShouldDeepEqual(new Foo
            {
                Bar = "test"
            });
        }

        [Fact]
        [UseReporter(typeof(VsCodeReporter))]
        public void ApprovalWorks()
        {
            var foo = new Foo { Bar = "test" };
            Approvals.Verify(foo);
        }

        class Foo
        {
            public String Bar { get; set; }

            public override String ToString()=>
                $"Foo(Bar={Bar})";
        }
    }
}

File: fsharp/gol.tests/TestTheTester.fs
module TestTheTester

open System
open Xunit
open FsCheck.Xunit
open Swensen.Unquote

[<Fact>]
let ``It works`` () =
    Assert.Equal(1, 1)

[<Fact>]
let ``Unquote works`` () =
    test <@ 1 = 1 @>

[<Property>]
let ``Reverse of reverse of a list is the original list ``(xs:list<int>) =
    List.rev(List.rev xs) = xs